src/app/frontend/common/services/global/history.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright 2017 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Injectable, transition} from '@angular/core';
+import {RawParams, StateService, Transition, TransitionPromise, TransitionService} from '@uirouter/core';
+import {KdStateService} from './state';
+
+@Injectable()
+export class HistoryService {
+  private previousStateName: string;
+  private previousStateParams: RawParams;
+
+  constructor(private readonly state: StateService, private readonly kdState: KdStateService) {
+    this.init();
+  }
+
+  /** Initializes the service. Must be called before use. */
+  init(): void {
+    this.kdState.onSuccess.subscribe((transition: Transition) => {
+      this.previousStateName = transition.from().name || '';
+      this.previousStateParams = transition.params('from');
+    });
+  }
+
+  /**
+   * Goes back to previous state or to the provided default if none set.
+   * @param {string} defaultState
```

Comment:
We do not longer use type annotations in comments.

Reply:
Moreover, `back` doesn't explain what the function does.

Reply:
will update the changes in #2907. `back` will be changed to 'goBack'

Reply:
`goBack` or `goToPreviousState` seem to be okay for me.

cc @floreks